payjoin/src/uri.rs

```diff
@@ -274,11 +312,52 @@ enum InternalPjParseError {
     UnsecureEndpoint,
 }
 
+pub struct PjUriBuilder<'a> {
+    address: Address,
+    pj_endpoint: Url,
+    amount: Option<Amount>,
+    ohttp: Option<&'a str>,
+}
+
+impl<'a> PjUriBuilder<'a> {
+    pub fn new(address: Address, pj_endpoint: Url) -> Self {
+        Self { address, pj_endpoint, amount: None, ohttp: None }
+    }
+
+    pub fn ohhtp(mut self, ohttp: Option<&'a str>) -> Self {
+        self.ohttp = ohttp;
+        self
+    }
+
+    pub fn amount(mut self, amount: Option<bitcoin::Amount>) -> Self {
+        self.amount = amount;
```

Comment:
I understand your point, and I was hesitant first about this, but decided to add it because:
1) if you have a function that accepts Option< Amount >, like https://github.com/payjoin/rust-payjoin/pull/148/files#diff-8e6acd92d1374811117a8f9f08387d1a66bd8a3d2ae69c669dd65bab00e441e5L431 - if we make amount non optional in the .amount function it will make it ugly to call our function
2) i think its better to always match the input to the actual type in the struct
3) we can think about the functions as set/unset and not just set

I dont feel strongly about this, but also not a big fan of making it non optional. Third option could be to have only 2 functions in the struct, `new` and `build` and new would take another two Option< amount > Option< str > args.

Reply:
Wise review. I appreciate your thought process. A `PjUrl` constructor might make more sense than a builder in our impl. I figured a bip21 builder trait would fit better with with other params like `lightning` that are outside of the scope of this but I have a feeling that's actually a job of crates like [bitcoin-waila](https://github.com/MutinyWallet/bitcoin-waila) and support from `bip21`.